src/main/kotlin/com/autonomousapps/internal/advice/AdvicePrinter.kt

```diff
@@ -46,6 +48,26 @@ internal class AdvicePrinter(
         }  }}"
       }
     }
+    return dependency
+  }
+
+  private fun getProjectFormat(quotedDep: String): String {
+    return if (useTypesafeProjectAccessors) {
+      if (dslKind == DslKind.KOTLIN) {
+        "projects${quotedDep.replace(':', '.').replace("\"", "").kebabToCamelCase()}"
+      } else {
+        "projects${quotedDep.replace(':', '.').replace("'", "").kebabToCamelCase()}"
+      }
+    } else {
+      if (dslKind == DslKind.KOTLIN) "project(${quotedDep})" else "project(${quotedDep})"
+    }
+  }
+
+  private fun String.kebabToCamelCase(): String {
+    val pattern = "-[a-z]".toRegex()
```

Comment:
Does this regex work in all possible cases? Does gradle only permit ASCII characters or is a wider character range available?